xonsh/main.py

```diff
@@ -51,12 +51,12 @@ def path_argument(s):
                     dest='command',
                     required=False,
                     default=None)
-parser.add_argument('-i',
+parser.add_argument('-i', '--interactive',
                     help='force running in interactive mode',
                     dest='force_interactive',
                     action='store_true',
                     default=False)
-parser.add_argument('-l',
+parser.add_argument('-l', '--login',
```

Comment:
Basically, they add these elements back in to the `args` or `argv` list.  This is needed if you are writing a command line script written in xonsh, ie `#!/usr/bin/env xonsh`.  The way argparse works is that it consumes these and thus they won't necessarily get passed down to the utility.  

Say that you were writing a command `test`, running `test -h` would give you xonsh's help NOT test's help. The arg undoer mitigate this.